src/status_im/hardwallet/core.cljs

```diff
@@ -45,9 +45,42 @@
 
 (defn return-back-from-nfc-settings [app-coming-from-background? {:keys [db]}]
   (when (and app-coming-from-background?
-             (= :hardwallet/connect (:view-id db)))
+             (= :hardwallet-connect (:view-id db)))
     {:hardwallet/check-nfc-enabled nil}))
 
+(defn- proceed-to-pin-confirmation [fx]
+  (assoc-in fx [:db :hardwallet :pin :enter-step] :confirmation))
+
+(defn- pin-match [fx]
+  (assoc-in fx [:db :hardwallet :pin :status] :validating))
+
+(defn- pin-mismatch [fx]
+  (-> fx
+      (assoc-in [:db :hardwallet :pin :status] :error)
+      (assoc-in [:db :hardwallet :pin :error] :t/pin-mismatch)
+      (assoc-in [:db :hardwallet :pin :original] [])
+      (assoc-in [:db :hardwallet :pin :confirmation] [])
+      (assoc-in [:db :hardwallet :pin :enter-step] :original)))
```

Comment:
It would be probably easier on eyes to just merge map into `[:db :hardwallet :pin]` like this:
```clj
(update-in fx [:db :hardwallet :pin] merge {:status       :error
                                            :error        :t/pin-mismatch
                                            :original     []
                                            :confirmation []
                                            :enter-step   :original})
```
But that's a minor point and opinion piece anyway

Reply:
@janherich good idea actually, looks simpler.